clar2wasm/tests/wasm-generation/tokens.rs

```diff
@@ -110,3 +110,125 @@ proptest! {
+                ),
+            ])
+            .unwrap(),
+        );
+
+        crosscheck(&snippet, Ok(Some(expected)));
+    }
+
+    #[test]
+    fn ft_mint_burn(
+        total_supply in any::<u128>(),
+        recipient in PropValue::from_type(PrincipalType),
+    ) {
+        let snippet = format!(r#"
+            (define-fungible-token stackaroo u{total_supply})
+            {{
+                a-mint: (ft-mint? stackaroo u{total_supply} {recipient}),
+                b-burn: (ft-burn? stackaroo u{total_supply} {recipient}),
+                c-balance: (ft-get-balance stackaroo {recipient}),
+            }}
+        "#);
+
+        let expected = Value::from(
+            TupleData::from_data(vec![
+                (
+                    ClarityName::from("a-mint"),
+                    Value::okay_true(),
+                ),
+                (
+                    ClarityName::from("b-burn"),
+                    Value::okay_true(),
+                ),
+                (
+                    ClarityName::from("c-balance"),
+                    Value::UInt(0),
+                ),
+            ])
+            .unwrap(),
+        );
+
+        crosscheck(&snippet, Ok(Some(expected)));
+    }
+
+    #[test]
+    fn ft_mint_mint_supply_transfer_balance(
+        total_supply in any::<u128>(),
+        sender in PropValue::from_type(PrincipalType),
+        recipient in PropValue::from_type(PrincipalType),
+    ) {
+        let mint_supply = total_supply >> 1;
+
+        let snippet = format!(r#"
+            (define-fungible-token stackaroo u{total_supply})
+            {{
+                a-mint1: (ft-mint? stackaroo u{mint_supply} {sender}),
+                b-mint2: (ft-mint? stackaroo u{mint_supply} {recipient}),
+                c-supply: (ft-get-supply stackaroo),
+                d-transfer: (ft-transfer? stackaroo u{mint_supply} {sender} {recipient}),
+                e-balance-sender: (ft-get-balance stackaroo {sender}),
+                f-balance-recipient: (ft-get-balance stackaroo {recipient}),
+            }}
+        "#);
+
+        let expected = Value::from(
+            TupleData::from_data(vec![
+                (
+                    ClarityName::from("a-mint1"),
+                    Value::okay_true(),
+                ),
+                (
+                    ClarityName::from("b-mint2"),
+                    Value::okay_true(),
+                ),
+                (
+                    ClarityName::from("c-supply"),
+                    Value::UInt(total_supply & !1),
+                ),
+                (
+                    ClarityName::from("d-transfer"),
+                    Value::okay_true(),
+                ),
+                (
+                    ClarityName::from("e-balance-sender"),
+                    Value::UInt(0),
+                ),
+                (
+                    ClarityName::from("f-balance-recipient"),
+                    Value::UInt(mint_supply * 2),
```

Comment:
Considering this code will be executed multiple times, it might be worth changing it to `Value::UInt(total_supply)` to avoid a multiplication operation in a u128 value.

Reply:
This is not correct! We have an integer division, so it could be `total_supply` only if it's even. Also, a multiplication by 2 translates to a single instruction in machine code, and it should be a bit shift.